pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -868,13 +923,14 @@ func KubeStateMetricsFactoryWithParam(labelsMapper map[string]string, labelJoins
 
 func newKSMCheck(base core.CheckBase, instance *KSMConfig) *KSMCheck {
 	return &KSMCheck{
-		CheckBase:          base,
-		instance:           instance,
-		telemetry:          newTelemetryCache(),
-		isCLCRunner:        ddconfig.IsCLCRunner(),
-		metricNamesMapper:  defaultMetricNamesMapper(),
-		metricAggregators:  defaultMetricAggregators(),
-		metricTransformers: defaultMetricTransformers(),
+		CheckBase:            base,
+		instance:             instance,
+		telemetry:            newTelemetryCache(),
+		isCLCRunner:          ddconfig.IsCLCRunner(),
+		isRunningOnNodeAgent: flavor.GetFlavor() != flavor.ClusterAgent && !ddconfig.IsCLCRunner(),
```

Comment:
since we have already `isCLCRunner` field should we store instead `isClusterAgent` to simplify the operation?

Or replace it by an `agentFlavor` field, that will contains the flavor (agent, agent-clc, cluster-agent)?

Reply:
We can't use only the flavor, because there isn't a flavor for the CLC: https://github.com/DataDog/datadog-agent/blob/80efde26183428ecf06965a3d6799864a5c71126/pkg/util/flavor/flavor.go#L11

I don't think that storing `isClusterAgent` instead of `isRunningOnNodeAgent` simplifies things because then I'd need to use `!isClusterAgent && !isCLCRunner` in a few places.

Reply:
sounds good, this is what I saw after too